README.md

```diff
@@ -21,38 +21,77 @@ npm install --save aws-elasticsearch-connector @elastic/elasticsearch aws-sdk
 ### Using global configuration
 
 ```javascript
-const { Client } = require('@elastic/elasticsearch')
-const AWS = require('aws-sdk')
-const createAwsElasticsearchConnector = require('aws-elasticsearch-connector')
+const { Client } = require("@elastic/elasticsearch");
+const AWS = require("aws-sdk");
+const createAwsElasticsearchConnector = require("aws-elasticsearch-connector");
 
 // (Optional) load profile credentials from file
 AWS.config.update({
-  profile: 'my-profile'
-})
+  profile: "my-profile",
+});
 
 const client = new Client({
   ...createAwsElasticsearchConnector(AWS.config),
-  node: 'https://my-elasticsearch-cluster.us-east-1.es.amazonaws.com'
-})
+  node: "https://my-elasticsearch-cluster.us-east-1.es.amazonaws.com",
+});
 ```
 
 ### Using specific configuration
 
 ```javascript
-const { Client } = require('@elastic/elasticsearch')
-const AWS = require('aws-sdk')
-const createAwsElasticsearchConnector = require('aws-elasticsearch-connector')
+const { Client } = require("@elastic/elasticsearch");
+const AWS = require("aws-sdk");
+const createAwsElasticsearchConnector = require("aws-elasticsearch-connector");
 
 const awsConfig = new AWS.Config({
   // Your credentials and settings here, see
   // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#constructor-property
-})
+});
 
 const client = new Client({
   ...createAwsElasticsearchConnector(awsConfig),
-  node: 'https://my-elasticsearch-cluster.us-east-1.es.amazonaws.com'
-})
-````
+  node: "https://my-elasticsearch-cluster.us-east-1.es.amazonaws.com",
+});
+```
+
+### Using aws-sdk v3
+
+```javascript
+const { STSClient, AssumeRoleCommand } = require("@aws-sdk/client-sts");
+const { Client } = require("@elastic/elasticsearch");
+const createAwsElasticsearchConnector = require("./src/index.js");
+
+async function ping() {
+  const creds = await assumeRole(
+    "arn:aws:iam::062437324875:role/Administrator",
```

Comment:
I don't think it's safe to include this ARN which contains your AWS ID here in the example. Please change it to an obviously fake number, or better, demonstrate in the example obtaining this from a source that is safe for storing secrets.